packages/components/src/components/common/table/hooks/hooks.tsx

```diff
@@ -200,7 +201,7 @@ export const useTableKeyHandler = <TData,>({ table, data, options }: TableKeyboa
       table.resetRowSelection();
       allRows[newReorderIndex].toggleSelected();
       setRootIndex(newReorderIndex);
-      event.currentTarget.rows[newReorderIndex].scrollIntoView({ block: 'center' });
```

Comment:
the problem here was that event.currentTarget.rows only contains the virtualized rows. So as soon as there are rows at the start which only exists in the memory but not as dom nodes the index is no longer matching.

The new solution checks if there is a data-vindex attribute on the row and tries to get the correct row via this attribute.